receiver/mongodbreceiver/scraper.go

```diff
@@ -90,9 +90,20 @@ func (s *mongodbScraper) collectMetrics(ctx context.Context, errors scrapererror
 
 	now := pcommon.NewTimestampFromTime(time.Now())
 
+	s.mb.RecordMongodbDatabaseCountDataPoint(now, int64(len(dbNames)))
 	s.collectAdminDatabase(ctx, now, errors)
+	s.collectTopStats(ctx, now, errors)
 	for _, dbName := range dbNames {
 		s.collectDatabase(ctx, now, dbName, errors)
+		collectionNames, err := s.client.ListCollectionNames(ctx, dbName)
```

Comment:
Ran into some issues with the LPU recommendation with `clusterMonitor` permissions here. We may want to update documentation guidance in order to collect these. There may need to be some discussion if we want to elevate the type of user recommended for collection. 

the error:
```
2022-07-25T14:56:50.491-0400	error	mongodbreceiver@v0.56.0/scraper.go:100	Failed to fetch collection names	{"kind": "receiver", "name": "mongodb", "pipeline": "metrics", "error": "(Unauthorized) not authorized on admin to execute command { listCollections: 1, filter: {}, nameOnly: true, cursor: {}, lsid: { id: UUID(\"ba7d5c19-803f-4d2f-b7f3-1338116d88bb\") }, $db: \"admin\" }"}
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).collectMetrics
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver@v0.56.0/scraper.go:100
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).scrape
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver@v0.56.0/scraper.go:80
go.opentelemetry.io/collector/receiver/scraperhelper.ScrapeFunc.Scrape
	go.opentelemetry.io/collector@v0.56.0/receiver/scraperhelper/scraper.go:32
go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).scrapeMetricsAndReport
	go.opentelemetry.io/collector@v0.56.0/receiver/scraperhelper/scrapercontroller.go:194
go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).startScraping.func1
```

command used to create the user in question.
```
db.createUser({ user: "mUser", pwd: "otel", roles: [ { role: "clusterMonitor", db: "admin" } ]})
Successfully added user: {
	"user" : "mUser",
	"roles" : [
		{
			"role" : "clusterMonitor",
			"db" : "admin"
		}
	]
}
```

Reply:
Looking into this more because metrics seem useful

Reply:
Already talked to @JonathanWamsley but for observability, just want to clarify what I see as options at the current moment.

The issue revolves around collecting information about the admin database's collections. 

- Option 1 (breaking change):
  - We elevate the monitoring user we recommend with more permissions to be somewhat analagous to an read only `admin` user and update corresponding documentation so that it can collect against the admin database.
  - This may take more research but need to find the correct permissions to perform the proposed functionality.


- Option 2 (non-breaking change): 
  - Attempt collection and try to scrape everything with the permissions we currently have, if it fails then we perhaps log something or return a partial error on first scrape?
  - The main downside is how to communicate in documentation a blurry line to document a class of metrics that require more permissions than documented LPU.

My opinion is that we should try to aim for option 1, but I'm not entirely sure which is exactly preferred from a user perspective. There may be more options available but these were the two that came to me first. Curious if you have any thoughts @djaglowski?

Reply:
I'm not super familiar with the metrics in question but unless they would be sorely missed, I think we should:
1. Gracefully handle the case where we cannot collect them. Possibly this means requiring a config flag to enable collection of these metrics.
2. Suggest the lesser permissions as LPU.
3. Clearly document which additional permissions are necessary to collect additional metrics.

WDYT @schmikei @JonathanWamsley?